crates/rpc/rpc-types/src/eth/trace/geth/pre_state.rs

```diff
@@ -30,14 +38,13 @@ pub struct AccountState {
     pub balance: Option<U256>,
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub code: Option<Bytes>,
-    #[serde(
-        default,
-        deserialize_with = "from_int_or_hex_opt",
-        skip_serializing_if = "Option::is_none"
-    )]
-    pub nonce: Option<U256>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub nonce: Option<u64>,
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub storage: Option<BTreeMap<H256, H256>>,
+
+    #[serde(skip_serializing)]
+    pub change_type: ChangeType,
```

Comment:
I don't like this, this isn't part of the rpc object and should not be here, because this breaks serde symmetry even though changetype is just internally.

I suggest a helper type that wraps AccountState and changetype